### PR TITLE
Okta 347728 fetch openapi from npm

### PR DESCRIPTION
**Pull request by VitaliiTytarenko-okta**
## Story
[OKTA-347728](https://oktainc.atlassian.net/browse/OKTA-347728)

## Description
Presently the Java Mgmt SDK is using a local copy of the Open API spec. It was originally designed that way and should be changed to use the one from NPM.

We should update the SDK to fetch the Open API spec from npm package @okta/openapi directly like other management SDKs do today.

Note: Presently, the openapi repo is private and is in the process of getting moved to public.

Use VM option: `-Dexec.skip=true` to skip this fetching.

## Category
- [ ] Bugfix
- [x] Enhancement
- [ ] New Feature
- [x] Configuration Change
- [ ] Versioning Change
- [ ] Unit Test(s)
- [ ] Documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "@okta/openapi": "2.7.0"
```
Comment (arvindkrishnakumar-okta) [inline package.json:3]: @VitaliiTytarenko-okta please set it to `2.6.0`, the SDK has not been migrated to `2.7.0` yet.

**Comment by arvindkrishnakumar-okta** [comment]
@VitaliiTytarenko-okta 

Local build shows the below error:

```
[INFO] --- exec-maven-plugin:1.6.0:exec (npm install (initialize)) @ okta-sdk-api ---
env: node: No such file or directory
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 127 (Exit value: 127)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Okta Java SDK 5.0.1-SNAPSHOT:
[INFO]
[INFO] Okta Java SDK ...................................... SUCCESS [  3.242 s]
[INFO] Okta Java SDK :: Swagger Templates ................. SUCCESS [  7.280 s]
[INFO] Okta Java SDK :: API ............................... FAILURE [  0.197 s]
[INFO] Okta Java SDK :: Impl .............................. SKIPPED
[INFO] Okta Java SDK :: Extensions :: HTTP Client ......... SKIPPED
[INFO] Okta Java SDK :: Extensions :: OkHttp Client ....... SKIPPED
[INFO] Okta Java SDK :: Integration Tests ................. SKIPPED
[INFO] Okta Java SDK :: Examples .......................... SKIPPED
[INFO] Okta Java SDK :: Examples :: Quickstart Code ....... SKIPPED
[INFO] Okta Java SDK :: Coverage .......................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

**Comment by arvindkrishnakumar-okta** [comment]
@VitaliiTytarenko-okta since we now require npm to build this project, we must add a note in README or at https://github.com/okta/okta-sdk-java/wiki/Build-It with this requirement.

File: .gitignore [hunk 2]
```diff
@@ -47,3 +47,6 @@ npm-debug.log
 travis_phantomjs
 .DS_Store
 .vscode
+
+# api.yaml will be fetched from NPM
```
Comment (arvindkrishnakumar-okta) [inline .gitignore:51]:
```suggestion
# api.yaml will be fetched from NPM, the local copy will not be used.
```

Comment (arvindkrishnakumar-okta) [reply .gitignore:51]: nit: @VitaliiTytarenko-okta

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Review by sergiishamrai-okta — APPROVED** [review]

**Pull request merged**